core/src/test/java/kafka/test/junit/ClusterTestExtensions.java

```diff
@@ -119,7 +139,27 @@ public Stream<TestTemplateInvocationContext> provideTestTemplateInvocationContex
         return generatedContexts.stream();
     }
 
+    @Override
+    public void beforeEach(ExtensionContext context) {
+        DetectThreadLeak detectThreadLeak = DetectThreadLeak.of(thread ->
+                SKIPPED_THREAD_PREFIX.stream().noneMatch(prefix -> thread.getName().startsWith(prefix)));
+        getStore(context).put(DETECT_THREAD_LEAK_KEY, detectThreadLeak);
+    }
+
+    @Override
+    public void afterEach(ExtensionContext context) throws InterruptedException {
+        DetectThreadLeak detectThreadLeak = getStore(context).remove(DETECT_THREAD_LEAK_KEY, DetectThreadLeak.class);
+        if (detectThreadLeak == null) {
+            return;
+        }
+        TestUtils.waitForCondition(() -> detectThreadLeak.newThreads().isEmpty(),
+                "Thread leak detected: " +
```

Comment:
We need to display the threads from "last" call rather than the "initial" call.
```java
        AtomicReference<List<Thread>> lastThread = new AtomicReference<>();
        TestUtils.waitForCondition(() -> {
                List<Thread> threads = detectThreadLeak.newThreads();
                lastThread.set(threads);
                return threads.isEmpty();
            }, () -> "Thread leak detected: "
                + lastThread.get().stream().map(Thread::getName).collect(Collectors.joining(", ")));
```